Meta: Add templates/checklists for new, Accepted/Rejected & Final PEPs (#2956)
python/peps: .github/PULL_REQUEST_TEMPLATE/mark-pep-final.md

---
name: Mark PEP Final
about: Mark an Accepted Standards Track PEP as Final

> title: PEP NNN: Mark Final

Reviewer: ```suggestion
title: "PEP NNN: Mark Final"
```

---
name: Mark PEP Final
about: Mark an Accepted PEP as Final
---

<!--

See PEP 1 for more about the PEP lifecycle and when one is considered Final:

https://peps.python.org/pep-0001/#pep-review-resolution

Make sure to include the PEP number in the pull request title; for example:

PEP NNN: Mark Final

Thanks!

-->

# Mark PEP Final

> 
> <!--
> 
> State why the PEP should be marked Final,
> linking any relevant discussions, issues or PRs.
> 
> -->
> 
> 
> 
> ## Checklist
> 
> <!--
> 
> This lists the key steps needed to mark a PEP as Final.

Reviewer 2: ```suggestion

<!--

This checklist contains the key changes needed to mark a PEP as Final.
```

You can help complete it yourself if you like
by ticking any boxes you're sure about, like this: [x]
If you're unsure about something, just leave it blank and we'll take a look.

-->

> * [ ] Specification has been implemented in a released stable CPython version (or third-party tools, for packaging/typing topic PEPs)
> * [ ] ``Status`` changed to ``Final``
> * [ ] Canonical docs/spec moved to an appropriate place, and linked with a ``canonical-doc`` directive (or ``pypa-spec``, for packaging PEPs)
> * [ ] PEP headers and content updated with any substantial changes during the implementation phase

Author: ```suggestion
* [ ] Final implementation has been merged (including tests and documentation)
* [ ] PEP matches the final implementation
* [ ] Any substantial changes since the accepted version approved by the SC/PEP delegate
* [ ] ``Status`` changed to ``Final``
* [ ] Canonical docs/spec linked with a ``canonical-doc`` directive (or ``pypa-spec``, for packaging PEPs)
```

I've adopted a revised form of your suggestion above, with some further trimming down

> IMO the canonical-spec point should link to a devguide page that documents the directives.

We certainly should document it somewhere (in a separate issue/PR). Right now it is PEP 12 that documents all the rest of the syntax, directives and formatting for PEPs, so presumably it should go there for now, rather than the devguide which doesn't currently contain any other PEP-specific content. (As mentioned previously a potential future move to the devguide could make sense, but this should be discussed, decided upon and implemented for all the similar information as opposed to piecemeal, resulting in a fragmentary experience for authors and PEP editors).
